### DISCUSSION TITLE
Proposal: modify nargo to support third party plugins like cargo

### DISCUSSION THREAD

**Discussion by Ethan-000** [noir-lang/noir#3383] [Ideas]
### Problem

Support third party plugins so nargo functionalities can be extended by the community without requiring the nargo team to maintain everything

more on this here https://github.com/rust-lang/cargo/wiki/Third-party-cargo-subcommands

related to https://github.com/noir-lang/noir/issues/2164 but should be more general

### Happy Case

users can run something like `nargo halo2 prove`

### Would you like to submit a PR for this Issue?

Maybe

**Comment by Savio-Sou** [comment]
@TomAFrench any thoughts on approaching https://github.com/noir-lang/noir/issues/2164 via this method?

**Comment by kevaundray** [comment]
If this is the way we want to move forward long term, we should go for this after having an MVP of dynamic backends since dynamic backends is a blocker for us moving to barretenberg master and using the release artifacts from barretenberg

**Comment by TomAFrench** [comment]
Yeah, changing to a generic plugin model would be a massive scope creep.

**Comment by Savio-Sou** [comment]
Makes sense, thanks for the clarifications!

**Comment by Savio-Sou** [comment]
Note that for integrating `halo2` / proving backend specifically, we now have [`nargo backend`](https://github.com/noir-lang/docs/issues/369) that supports installation and switching between backends.
